server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetFieldMappingAction.java

```diff
@@ -68,20 +63,15 @@ public String getName() {
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         final String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
-        final String[] types = request.paramAsStringArrayOrEmptyIfAll("type");
         final String[] fields = Strings.splitStringByCommaToArray(request.param("fields"));
 
-        boolean includeTypeName = request.paramAsBoolean(INCLUDE_TYPE_NAME_PARAMETER, DEFAULT_INCLUDE_TYPE_NAME_POLICY);
-        if (includeTypeName == false && types.length > 0) {
-            throw new IllegalArgumentException("Types cannot be specified unless include_type_name" +
-                " is set to true.");
-        }
         if (request.hasParam(INCLUDE_TYPE_NAME_PARAMETER)) {
+            request.paramAsBoolean(INCLUDE_TYPE_NAME_PARAMETER, DEFAULT_INCLUDE_TYPE_NAME_POLICY);
```

Comment:
not using the return value?

Reply:
This is intentional. The include_type_name parameter still needs to be read so we don't throw an error saying that it's an unknown parameter. Although, since at this point we expect all uses of this to have `include_typeName = true` maybe we should check the value and throw an exception if the value is false?

Reply:
+1